docs/DesignDocs/SGX_QuoteEx_Integration.md

```diff
@@ -0,0 +1,127 @@
+- The quote-ex library supports generation of SGX quote in multiple formats (including ECDSA-p256 and EPID variations). With quote-ex, quote generation is always done out-of-process by working with a background service (called AESM) on the local platform.
+
+An SGX platform can have either the quote-ex library or the DCAP library, or both of them installed. When both libraries are installed, the quote-ex library takes precedence, as it supports more evidence formats.
+
+The OE SDK host-side plugin library is changed to dynamically detect the presence of the two libraries, load one of them, and record the entry points of the loaded library for invocation in the flows described below.
+
+#### Alternative: Only Use the quote-ex Library
+
+Run-time dynamic detection and loading of multiple shared libraries complicates implementation. It also increases risk of version mismatch between OE SDK and the dependent libraries that is not detected at build time via library headers.
+
+As described previously, the quote-ex library supports a superset of formats as compared to the DCAP library, though it always depends on a background service for quote generation. If on SGX platforms the OE SDK always installs with the AESM background service (as a hard dependency), then an alternative implementation is to use the libsgx-quote-ex library only (instead of the DCAP library). With this implementation, the libsgx-quote-ex library is linked to the OE SDK at build time.
+
+### Support of SGX Evidence Formats Enumeration
+
+The SGX plugin code file `enclave/sgx/attester.c` implements the OE SDK API `oe_sgx_get_attesters()`. The implementation enumerates all supported SGX evidence formats, creates a list of attester plugins for them, and returns the created list to the caller.
+
+For SGX evidence formats enumeration, a new OCALL is added to interface definition file `common/sgx/sgx.edl` and implemented in the host-side SGX plugin library:
+
+- `oe_get_supported_attester_format_ids_ocall(void* format_ids, size_t format_ids_size, size_t* format_ids_size_out)`
+    - This OCALL returns a list of supported evidence format IDs in caller-supplied buffer, and returns the size of the buffer actually used to hold the list.
+    - But if the supplied buffer is missing or not large enough, it only returns the needed buffer size.
+
+In the implementation of this OCALL by the host-side SGX plugin library:
+
+- If the DCAP library is loaded, a list with a single evidence format ID for ECDSA-p256 is returned.
+- Otherwise if the quote-ex library is loaded, its API `sgx_get_supported_att_key_ids()` is invoked, and the returned list of attestation key IDs is converted to a list of OE SDK evidence format IDs.
+
+### Updated Implementation of OE SDK API oe_get_evidence()
```

Comment:
is this about the implementation of oe_get_evidence or about the implementatoin of get_evidence in an SGX plugin?   I am thinking it should be the latter, in which case it would be helpful to update the title in line 101

Reply:
Agreed. All changes are within the implementation of get_evidence() in an SGX plugin